src/components/Loading.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useEffect, useState } from 'react';
```

Comment:
I like this but my first impression was that our React version did not support this (https://reactjs.org/docs/hooks-reference.html). Support was added on 16.8.0 and our package.json uses 16.5.0.

However I was looking at package-lock.json and, for some reason, React version is 16.12.0 the since Feb 2020. I think it's safe to assume v16.12.0 works, just think we should upgrade the version in package.json as well